src/modules/cpu/kernel/fmadd_scalar.hpp

```diff
@@ -0,0 +1,126 @@
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#include "rppdefs.h"
+#include "rpp_cpu_simd.hpp"
+#include "rpp_cpu_common.hpp"
+
+RppStatus fmadd_scalar_f32_f32_host_tensor(Rpp32f *srcPtr,
+                                           RpptGenericDescPtr srcGenericDescPtr,
+                                           Rpp32f *dstPtr,
+                                           RpptGenericDescPtr dstGenericDescPtr,
+                                           Rpp32f *mulTensor,
+                                           Rpp32f *addTensor,
+                                           RpptROI3DPtr roiGenericPtrSrc,
+                                           RpptRoi3DType roiType,
+                                           RppLayoutParams layoutParams,
+                                           rpp::Handle& handle)
+{
+    RpptROI3D roiDefault = {0, 0, 0, (Rpp32s)srcGenericDescPtr->dims[3], (Rpp32s)srcGenericDescPtr->dims[4], (Rpp32s)srcGenericDescPtr->dims[2]};
+    Rpp32u numThreads = handle.GetNumThreads();
+
+    omp_set_dynamic(0);
+#pragma omp parallel for num_threads(numThreads)
+    for(int batchCount = 0; batchCount < dstGenericDescPtr->dims[0]; batchCount++)
+    {
+        RpptROI3D roi;
+        RpptROI3DPtr roiPtrInput = &roiGenericPtrSrc[batchCount];
+        compute_roi3D_validation_host(roiPtrInput, &roi, &roiDefault, roiType);
+
+        Rpp32f mulParam = mulTensor[batchCount];
+        Rpp32f addParam = addTensor[batchCount];
+
+        Rpp32f *srcPtrImage, *dstPtrImage;
+        srcPtrImage = srcPtr + batchCount * srcGenericDescPtr->strides[0];
+        dstPtrImage = dstPtr + batchCount * dstGenericDescPtr->strides[0];
+
+        Rpp32u bufferLength = roi.xyzwhdROI.roiWidth * layoutParams.bufferMultiplier;
+
+        Rpp32f *srcPtrChannel, *dstPtrChannel;
+        srcPtrChannel = srcPtrImage + (roi.xyzwhdROI.xyz.z * srcGenericDescPtr->strides[2]) + (roi.xyzwhdROI.xyz.y * srcGenericDescPtr->strides[3]) + (roi.xyzwhdROI.xyz.x * layoutParams.bufferMultiplier);
+        dstPtrChannel = dstPtrImage;
+
+
+#if __AVX2__
+        Rpp32u vectorIncrementPerChannel = 8;
+
+        __m256 pFmaddParams[2];
+        pFmaddParams[0] = _mm256_set1_ps(mulParam);
+        pFmaddParams[1] = _mm256_set1_ps(addParam);
+#endif
+        // Fmadd without fused output-layout toggle (NDHWC -> NDHWC or NCDHW -> NCDHW)
```

Comment:
Are you sure this will work for all?

Reply:
Nope this worked fine and matches with HIP as i was setting roi region wrongly, fixed it now.